File: C# OOP/Polymorphism - Lab/Shapes/Circle.cs
﻿using System;

namespace Shapes
{
    public class Circle : Shape
    {
        private double radius;

        public Circle(double radius)
        {
            Radius = radius;
        }

        public double Radius
        {
            get => radius;
            set
            {
                radius = value;
            }
        }

        public override double CalculateArea() => Math.PI * Math.Pow(Radius, 2);

        public override double CalculatePerimeter() => 2 * Math.PI * Radius;

        public override string Draw()
        {
            return base.Draw() + "Circle";
        }
    }
}

File: JS Advanced/Classes - Lab/circle.js
 class Circle {
    constructor(radius) {
        this._radius = radius;
    }

    get radius() { 
        return this._radius;
    }

    get diameter (){
        return this._radius * 2;
    }

    set diameter (value) {
        this._radius = value / 2;
    }

    get area(){
        return Math.PI * (this.radius ** 2)
    }
}

let c = new Circle(2);
console.log(`Radius: ${c.radius}`);
console.log(`Diameter: ${c.diameter}`);
console.log(`Area: ${c.area}`);

c.diameter = 1.6;
console.log(`Radius: ${c.radius}`);
console.log(`Diameter: ${c.diameter}`);
console.log(`Area: ${c.area}`);